### ISSUE TITLE
KinematicBody3D with capsule collision sliding along objects will cause crash

### ISSUE THREAD

**Issue by Prisalized** [godotengine/godot#40381]
**Godot version:**
v4.0.dev.custom_build.c2ed367d4

**OS/device including version:**
Windows 10

**Issue description:**
when a kinimaticbody3D has a capsule collision with the top sliding along a collision body it will crash the game window saying "!process_map->has(p_pid)" is true. Returning: FAILED

**Steps to reproduce:**
make a collision shape on kenimaticbody3D and make a collision shape with a capsule have it move upwards constantly and add moment with arrow keys when moving around wile the top of the capsule is colliding with something the game window will crash and give this error  platform\windows\os_windows.cpp:494 - Condition "!process_map->has(p_pid)" is true. Returning: FAILED

**Minimal reproduction project:**

[godot 4.0 collision bug.zip](https://github.com/godotengine/godot/files/4919387/godot.4.0.collision.bug.zip)

**Comment by akien-mga** [comment]
Confirmed on Linux. First there's a warning (happens directly when loading the project, not during the crash), then a Bullet assert. The crash is likely caused by #34236.

gdb backtrace:
```
warning btCollisionDispatcher::needsCollision: static-static collision!
godot-git: thirdparty/bullet/LinearMath/btVector3.h:305: btVector3& btVector3::normalize(): Assertion `!fuzzyZero()' failed.

Thread 1 "godot-git" received signal SIGABRT, Aborted.
0x00007ffff7788380 in raise () from /lib64/libc.so.6
(gdb) bt
#0  0x00007ffff7788380 in raise () from /lib64/libc.so.6
#1  0x00007ffff7773527 in abort () from /lib64/libc.so.6
#2  0x00007ffff777340f in __assert_fail_base.cold () from /lib64/libc.so.6
#3  0x00007ffff7780c42 in __assert_fail () from /lib64/libc.so.6
#4  0x00000000026c358e in btVector3::normalize (this=0x7fffffffa9c0) at thirdparty/bullet/LinearMath/btVector3.h:305
#5  0x00000000026c3ca0 in btVector3::normalized (this=0x7fffffffaa40) at thirdparty/bullet/LinearMath/btVector3.h:953
#6  0x000000000270d834 in btSingleSweepCallback::btSingleSweepCallback (this=0x7fffffffab10, castShape=0x91c7280, convexFromTrans=..., convexToTrans=..., world=0x9051190, resultCallback=..., 
    allowedPenetration=0.0399999991) at thirdparty/bullet/BulletCollision/CollisionDispatch/btCollisionWorld.cpp:1040
#7  0x0000000002709584 in btCollisionWorld::convexSweepTest (this=0x9051190, castShape=0x91c7280, convexFromWorld=..., convexToWorld=..., resultCallback=..., allowedCcdPenetration=0.0399999991)
    at thirdparty/bullet/BulletCollision/CollisionDispatch/btCollisionWorld.cpp:1101
#8  0x00000000026f6737 in SpaceBullet::test_body_motion (this=0x90424f0, p_body=0x91e2ff0, p_from=..., p_motion=..., p_infinite_inertia=true, r_result=0x7fffffffafd0, p_exclude_raycast_shapes=true)
    at modules/bullet/space_bullet.cpp:1011
#9  0x00000000026cc1f5 in BulletPhysicsServer3D::body_test_motion (this=0x8e0a620, p_body=..., p_from=..., p_motion=..., p_infinite_inertia=true, r_result=0x7fffffffafd0, p_exclude_raycast_shapes=true)
    at modules/bullet/bullet_physics_server.cpp:862
#10 0x0000000003a6212c in KinematicBody3D::move_and_collide (this=0x922c5e0, p_motion=..., p_infinite_inertia=true, r_collision=..., p_exclude_raycast_shapes=true, p_test_only=false)
    at scene/3d/physics_body_3d.cpp:915
#11 0x0000000003a625dd in KinematicBody3D::move_and_slide (this=0x922c5e0, p_linear_velocity=..., p_up_direction=..., p_stop_on_slope=false, p_max_slides=3, p_floor_max_angle=0.785398185, 
    p_infinite_inertia=true) at scene/3d/physics_body_3d.cpp:976
#12 0x0000000003a89506 in MethodBind6R<Vector3, Vector3 const&, Vector3 const&, bool, int, float, bool>::call (this=0x8b4ee30, p_object=0x922c5e0, p_args=0x7fffffffb738, p_arg_count=2, r_error=...)
    at ./core/method_bind_ext.gen.inc:309
#13 0x000000000464c0e6 in Object::call (this=0x922c5e0, p_method=..., p_args=0x7fffffffb738, p_argcount=2, r_error=...) at core/object.cpp:890
#14 0x00000000046dd72b in Variant::call_ptr (this=0x7fffffffb970, p_method=..., p_args=0x7fffffffb738, p_argcount=2, r_ret=0x0, r_error=...) at core/variant_call.cpp:1255
#15 0x00000000022b3eea in GDScriptFunction::call (this=0x88c6850, p_instance=0x91d0ef0, p_args=0x7fffffffcb38, p_argcount=1, r_err=..., p_state=0x0) at modules/gdscript/gdscript_function.cpp:1044
#16 0x0000000002263045 in GDScriptInstance::call_multilevel (this=0x91d0ef0, p_method=..., p_args=0x7fffffffcb38, p_argcount=1) at modules/gdscript/gdscript.cpp:1306
#17 0x00000000036a3914 in Node::_notification (this=0x922c5e0, p_notification=17) at scene/main/node.cpp:59
#18 0x0000000001ac7ca6 in Node::_notificationv (this=0x922c5e0, p_notification=17, p_reversed=false) at ./scene/main/node.h:46
#19 0x00000000020a0bd9 in Node3D::_notificationv (this=0x922c5e0, p_notification=17, p_reversed=false) at ./scene/3d/node_3d.h:52
--Type <RET> for more, q to quit, c to continue without paging--
#20 0x0000000003973187 in CollisionObject3D::_notificationv (this=0x922c5e0, p_notification=17, p_reversed=false) at ./scene/3d/collision_object_3d.h:38
#21 0x0000000003a7847b in PhysicsBody3D::_notificationv (this=0x922c5e0, p_notification=17, p_reversed=false) at scene/3d/physics_body_3d.h:41
#22 0x0000000003a79b19 in KinematicBody3D::_notificationv (this=0x922c5e0, p_notification=17, p_reversed=false) at scene/3d/physics_body_3d.h:258
#23 0x000000000464c203 in Object::notification (this=0x922c5e0, p_notification=17, p_reversed=false) at core/object.cpp:899
#24 0x00000000036d3fba in SceneTree::_notify_group_pause (this=0x770a3f0, p_group=..., p_notification=17) at scene/main/scene_tree.cpp:826
#25 0x00000000036d224c in SceneTree::idle (this=0x770a3f0, p_time=0.0166666675) at scene/main/scene_tree.cpp:455
#26 0x0000000001a6062f in Main::iteration () at main/main.cpp:2194
#27 0x0000000001a2ac14 in OS_LinuxBSD::run (this=0x7fffffffd260) at platform/linuxbsd/os_linuxbsd.cpp:238
#28 0x0000000001a2892c in main (argc=1, argv=0x7fffffffd768) at platform/linuxbsd/godot_linuxbsd.cpp:55
```

**Comment by madmiraal** [comment]
Odd. This should have been fixed with #34225.

**Comment by Prisalized** [comment]
thanks its now fixed for me
